src/main/java/com/suse/saltstack/netapi/config/SaltStackClientConfig.java

```diff
@@ -1,132 +1,92 @@
+    public static final Key<String> PROXY_PASSWORD = new Key<>();
 
     /**
-     * Default constructor.
+     * A key to use with {@link SaltStackClientConfig}.
+     * @param <T> The type of the value associated with this key.
      */
-    public SaltStackClientConfig() {
-        this.properties = new Properties();
-    }
+    static class Key<T> {
 
-    /**
-     * Sets a preference given by key and value. Use one of the public key strings above.
-     * NOTE: To set the URL, use method setURL instead.
-     *
-     * @param key
-     * @param value
-     */
-    public void put(String key, String value) {
-        // The URL key should be treated differently.
-        if (key.equals(SaltStackClientConfig.URL)) {
-            try {
-                this.setUrl(key);
-            } catch (URISyntaxException ex) {
-                // Complain to the log for invalid URL.
-                throw new RuntimeException(ex);
-            }
-        }
-        else {
-            properties.setProperty(key, value);
+        /** The default value of this key */
+        public final T defaultValue;
+
+        /**
+         * Creates a new Key with the default value null.
+         */
+        public Key(){
+          this(null);
         }
-    }
 
-    /**
-     * Removes a preference given by key.
-     *
-     * @param key
-     */
-    public void remove(String key) {
-        if (properties.containsKey(key)) {
-            properties.remove(key);
+        /**
+         * Creates a new key with the specified default value.
+         *
+         * @param defaultValue Default value for this key.
+         */
+        public Key(T defaultValue) {
+            this.defaultValue = defaultValue;
         }
-    }
 
-    /**
-     * Set the URL parameter.
-     *
-     * @param url
-     * @throws URISyntaxException
-     */
-    public void setUrl(String url) throws URISyntaxException {
-        properties.setProperty(URL, new URI(url).toASCIIString());
     }
 
-    /**
-     * Returns the currently used URL.
-     *
-     * @return url
-     */
-    public String getUrl() {
-        return properties.getProperty(URL, DEFAULT_URL);
-    }
+    private Map<Key, Object> store = new HashMap<>();
 
     /**
-     * Returns the proxy hostname or null.
+     *  Sets the config for a key to the specified value.
      *
-     * @return proxy hostname
+     * @param key The configuration key to set.
+     * @param value The value to associate with the key.
+     * @param <T> The type of the value associated with the key.
      */
-    public String getProxyHostname() {
-        return properties.getProperty(PROXY_HOSTNAME, null);
-    }
-
-     /**
-     * Returns the configured proxy port (or 3128 as the default).
-     *
-     * @return proxy port
-     */
-    public String getProxyPort() {
-        return properties.getProperty(PROXY_PORT, DEFAULT_PROXY_PORT);
+    @SuppressWarnings("unchecked")
```

Comment:
You are right its no longer needed in this implementation. A previous one did use some casting.
Using Java 8 didn't produce any warnings about it either.
Speaking of Java 8 is there any chance of upgrading to is or is 7 a hard requirement.
I'm just asking because it would be nice to use `Optional` instead of passing around `null` everywhere.